### ISSUE TITLE
how write es6  member?

### ISSUE THREAD

**Issue by izengliang** [jsdoc/jsdoc#819]
```
class Actor {

    /**
     * @constructor Actor
     */
    constructor() {

        /**
         * @member
         */
        this.name = "leo";

    }
}
```

how write `this.name` annotate ?

**Comment by mizzao** [comment]
Hey guys - is there an update on whether jsdoc supports ES6 now? I have seen some messages floating around but didn't see anything authoritative.

**Comment by mattvot** [comment]
Officially it looks to be in the low priority list with no planned release date.
https://github.com/kpi-wdc/wdc/issues/97

**Comment by hegemonic** [comment]
@mizzao @mattvot On GitHub master, JSDoc currently supports almost all ES2015 features. However, there has not been a formal release that includes ES2015 support, so the version on npm does not include ES2015 support.

JSDoc 3.4.0 will be the first formal release with ES2015 support. I don't have a firm date for that release; there are more changes to land, and my time has been scarce. For now, if you need ES2015 support, I recommend using the development version of JSDoc.

**Comment by mizzao** [comment]
Thanks! Good to know.

**Comment by JerryFox1** [comment]
@hegemonic which one is the dev one for es6 stuff?

**Comment by hegemonic** [comment]
@JerryFox1 GitHub master is the development version.

**Comment by mhagmajer** [comment]
@hegemonic, is the es6 classes support documented yet? I'm having trouble generating documentation for methods:

```javascript

/**
 * This is the main App class
 * @class App
 */
class App extends React.Component {
  /**
   * Constructs an instance od App.
   *
   * @param {Props} props - The props.
   * @constructs App
   */
  constructor(props) {
    super(props);
  }

  /**
   * Renders the component
   *
   * @method render
   */
  render() {
  }
}
```

"render" method does not show up in the docs. I'm on JSDoc 3.4.0-dev.

**Comment by tomauty** [comment]
@mhagmajer You want to specify that the `render` method is an instance method of `App`. This should work:
```javascript
/**
 * This is the main App class
 * @class App
 */
class App extends React.Component {
  /**
   * Constructs an instance od App.
   *
   * @param {Props} props - The props.
   * @constructs App
   */
  constructor(props) {
    super(props);
  }

  /**
   * Renders the component
   *
   * @method App#render
   */
  render() {
  }
}
```

**Comment by mhagmajer** [comment]
@tomauty, thanks for replying! Unfortunately this syntax doesn't seem render the method in the docs in `JSDoc 3.4.0-dev (Sun, 14 Jun 2015 18:49:07 GMT)`.

**Comment by tomauty** [comment]
Oh, yeah I'm currently using `3.3.0`. Looks like a change in 3.4 broke this functionality...

**Comment by popomore** [comment]
@hegemonic can you release 3.4 to npm with dev tag, because git is slow.

**Comment by davidreghay** [comment]
Exporting a class only after fully defining it seems to solve the problem of getting member _functions_ to render in the output docs, but with 3.4 I haven't found any way to get plain members to render any documentation.
